vortex-scan/src/range_scan.rs

```diff
@@ -103,11 +103,10 @@ impl RangeScan {
         match &self.state {
             State::FilterEval(_) => {
                 // Intersect the result of the filter expression with our initial row mask.
-                let mask = result.into_bool()?.boolean_buffer();
-                let mask = self.mask.to_boolean_buffer()?.bitand(&mask);
+                let mask = FilterMask::from_buffer(result.into_bool()?.boolean_buffer());
+                let mask = self.mask.clone().bitand(mask);
```

Comment:
it's weird that you have to do a clone here

Reply:
Yeah let's `impl BitAnd for &FilterMask` instead